src/frontend/src/optimizer/plan_node/logical_max_one_row.rs

```diff
@@ -0,0 +1,122 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use pretty_xmlish::XmlNode;
+use risingwave_common::bail;
+use risingwave_common::error::Result;
+
+use super::generic::DistillUnit;
+use super::utils::Distill;
+use super::{
+    gen_filter_and_pushdown, generic, ColPrunable, ExprRewritable, Logical, PlanBase, PlanRef,
+    PlanTreeNodeUnary, PredicatePushdown, ToBatch, ToStream,
+};
+use crate::optimizer::plan_node::expr_visitable::ExprVisitable;
+use crate::optimizer::plan_node::{
+    ColumnPruningContext, PredicatePushdownContext, RewriteStreamContext, ToStreamContext,
+};
+use crate::utils::{ColIndexMapping, Condition};
+
+/// [`LogicalMaxOneRow`] fetches up to one row from the input, returning an error
+/// if the input contains more than one row at runtime. Only available in batch mode.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct LogicalMaxOneRow {
+    pub base: PlanBase<Logical>,
+    pub(super) core: generic::MaxOneRow<PlanRef>,
+}
+
+impl LogicalMaxOneRow {
+    fn new(core: generic::MaxOneRow<PlanRef>) -> Self {
+        let base = PlanBase::new_logical_with_core(&core);
+        LogicalMaxOneRow { base, core }
+    }
+
+    pub fn create(input: PlanRef) -> PlanRef {
+        let core = generic::MaxOneRow { input };
+        Self::new(core).into()
+    }
+}
+
+impl PlanTreeNodeUnary for LogicalMaxOneRow {
+    fn input(&self) -> PlanRef {
+        self.core.input.clone()
+    }
+
+    fn clone_with_input(&self, input: PlanRef) -> Self {
+        let core = generic::MaxOneRow { input };
+        Self::new(core)
+    }
+
+    #[must_use]
+    fn rewrite_with_input(
+        &self,
+        input: PlanRef,
+        input_col_change: ColIndexMapping,
+    ) -> (Self, ColIndexMapping) {
+        (self.clone_with_input(input), input_col_change)
+    }
+}
+impl_plan_tree_node_for_unary! {LogicalMaxOneRow}
+
+impl Distill for LogicalMaxOneRow {
+    fn distill<'a>(&self) -> XmlNode<'a> {
+        self.core.distill_with_name("LogicalMaxOneRow")
+    }
+}
+
+impl ColPrunable for LogicalMaxOneRow {
+    fn prune_col(&self, required_cols: &[usize], ctx: &mut ColumnPruningContext) -> PlanRef {
+        let new_input = self.input().prune_col(required_cols, ctx);
+        self.clone_with_input(new_input).into()
+    }
+}
+
+impl ExprRewritable for LogicalMaxOneRow {}
+
+impl ExprVisitable for LogicalMaxOneRow {}
+
+impl PredicatePushdown for LogicalMaxOneRow {
+    fn predicate_pushdown(
+        &self,
+        predicate: Condition,
+        ctx: &mut PredicatePushdownContext,
+    ) -> PlanRef {
+        // Filter can not transpose MaxOneRow
+        gen_filter_and_pushdown(self, predicate, Condition::true_cond(), ctx)
+    }
+}
+
+impl ToBatch for LogicalMaxOneRow {
+    fn to_batch(&self) -> Result<PlanRef> {
+        todo!()
```

Comment:
todo?

Reply:
Oops. Forget to change this after reimplementation.